app/models/user.rb

```diff
@@ -4,7 +4,12 @@ class User < ApplicationRecord
   UPDATE_USERS_PARAMS = %i(name password password_confirmation).freeze
   attr_accessor :remember_token, :activation_token, :reset_token
   has_many :microposts, dependent: :destroy
-  before_save :downcase_email
+  has_many :active_relationships, class_name: Relationship.name, foreign_key: :follower_id, dependent: :destroy
+  has_many :passive_relationships, class_name: Relationship.name, foreign_key: :followed_id, dependent: :destroy
+  has_many :following, through: :active_relationships, source: :followed
+  has_many :followers, through: :passive_relationships, source: :follower
+  
+  before_save { email.downcase! }
```

Comment:
why you remove line before_save :downcase_email, why don't use it?

Reply:
error occured when i using before_save :downcase_email